internal/manager/feature_gates.go

```diff
@@ -0,0 +1,39 @@
+package manager
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+)
+
+// featureGatesDocsURL provides a link to the documentation for feature gates in the KIC repository
+const featureGatesDocsURL = "https://github.com/Kong/kubernetes-ingress-controller/blob/main/FEATURE_GATES.md"
+
+// setupFeatureGates converts feature gates to controller enablement
+func setupFeatureGates(setupLog logr.Logger, c *Config) (map[string]bool, error) {
```

Comment:
This function could have a simpler and cohesive mental model: take two `map[string]bool`s (rather than `c`) and have the semantics of "merging two maps key-wise but using the first in case of a conflict".